### PR TITLE
feat: allow child process instance migration

### LINKED ISSUE #18594: Support child instances in instance migration
### Description
Zeebe [supports the migration](https://github.com/camunda/zeebe/issues/15922) of child process instances. 
Operate needs to enable users to migrate child process instances. 

### User Story

```Gherkin
Scenario: As an Operate user
    Given I am in the process instance list
    When I select instances with parent process instances
    Then I can define the migration plan and migrate child instances
```
### Acceptance Criteria

- [ ] Child instances are selectable in the instance list to migrate
- [ ] I can define the migration plan for child instances
- [ ] I can execute migration for child instances
- [ ] [Docs](https://docs.camunda.io/docs/next/components/operate/userguide/process-instance-migration/) are updated: child instances are added to the list of mappable elements

### Definition of Ready - Checklist

- [x] The issue has a meaningful title, description, and testable acceptance criteria
- [ ] If documentation needs to be updated, an issue is created in the [camunda-docs](https://github.com/camunda/camunda-docs) repo, and the issue is added to our Operate project board.
- [ ] If HELM charts need to be updated, an issue is created in the [camunda-platform-heml](https://github.com/camunda/camunda-platform-helm) repo, and the issue is added to our Operate project board.
- [ ] Potential Security risks have been considered and added to the product-hub issue risk assessment
- [ ] Cross-team dependencies have been considered

Optional:
- [ ] Design input has been collected by the assignee
### :point_right: Handover Dev to QA 

- Handy resources: 
     BPMN/DMN models, plugins, scripts, REST API endpoints + example payload, etc : 
     
- Example projects:

- Versions to validate:
> Docker file : in case that it needed to be tested via docker share the version contain the fixed along with version of other services . 

- Release version ( in which version this fixed/feature will be released)

### PR DESCRIPTION
**Pull request by vitorwtavares**
## Description

- Allows migration for child process instances
- Adds new E2E test

## Related issues

closes #18594

### CODE REVIEW

File: operate/client/e2e-playwright/tests/childProcessInstanceMigration.spec.ts
Comment (pedesen) [inline operate/client/e2e-playwright/tests/childProcessInstanceMigration.spec.ts:154]: This is the point where the roundtrip happens (we're waiting for Zeebe to finish and for Operate to import the changes).

I saw the other migration test failing often in CI at this point, because the Operation had not been finished before the timeout (which I think is 30 seconds?). So I think it makes sense to increase the timeout here (and in processInstanceMigration.spec.ts too):

```suggestion
    await expect(
      migrateOperationEntry.getByRole('progressbar'),
    ).not.toBeVisible([timeout: 60000);
```

This will hopefully increase the stability a bit.

**Review by pedesen — COMMENTED** [review]
@vitorwtavares Great that you took the extra effort and created an E2E test for this! :medal_sports: 

It was hard to find something to complain about, I just added a comment related to CI stability. You can either fix it in this PR or in a separate one, I'll leave it up to you.

**Review by pedesen — APPROVED** [review]

File: operate/client/e2e-playwright/tests/childProcessInstanceMigration.spec.ts
Comment (vitorwtavares) [reply operate/client/e2e-playwright/tests/childProcessInstanceMigration.spec.ts:154]: This makes sense to me, I'll make the change in these 2 tests and also on `batchMoveModification`, that has an identical line and has already been flaky before too.

**Comment by vitorwtavares** [comment]
@Szik Please check using this image name `registry.camunda.cloud/team-operate/camunda-operate:branch-fe-18594-child-instances-migration`.

To test:
You can just try to migrate any child process (recommend using the `called-process` process since it's already called and you don't have to create anything new), if the "Migrate" button on the blue strip on the process instances list is enabled it should already work normally, since inside the migration mode no changes were made.

**Comment by aleksander-dytko** [comment]
@Szik could you please treat this issue with high priority? We have a customer waiting for this to be released in a patch release, still this week.

**Review by Szik — APPROVED** [review]

**Pull request merged**
